app/src/debug/java/org/mozilla/fenix/DebugFenixApplication.kt

```diff
@@ -8,20 +8,31 @@ import android.os.StrictMode
 import androidx.preference.PreferenceManager
 import leakcanary.AppWatcher
 import leakcanary.LeakCanary
+import org.mozilla.fenix.ext.application
 import org.mozilla.fenix.ext.getPreferenceKey
 
 class DebugFenixApplication : FenixApplication() {
 
     override fun setupLeakCanary() {
+        if (!AppWatcher.isInstalled) {
+            AppWatcher.manualInstall(
+                application = application,
+                watchersToInstall = AppWatcher.appDefaultWatchers(application)
+            )
+        }
+
         val isEnabled = components.strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
             PreferenceManager.getDefaultSharedPreferences(this)
                 .getBoolean(getPreferenceKey(R.string.pref_key_leakcanary), true)
         }
+
         updateLeakCanaryState(isEnabled)
     }
 
     override fun updateLeakCanaryState(isEnabled: Boolean) {
-        AppWatcher.config = AppWatcher.config.copy(enabled = isEnabled)
-        LeakCanary.config = LeakCanary.config.copy(dumpHeap = isEnabled)
+        LeakCanary.showLeakDisplayActivityLauncherIcon(isEnabled)
+        components.strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
```

Comment:
see https://github.com/square/leakcanary/issues/2117.
We could watch the issue and remove this reset after that has been solved.

Reply:
cc @mcomella

Reply:
Since it's in the debug fenix application, I'm not too concerned. Thanks for the heads up.